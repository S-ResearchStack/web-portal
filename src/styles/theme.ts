import { css, DefaultTheme } from 'styled-components';
import { px } from './utils';

const colors = {
  // gray gradations
  black: '#000000',
  black60: 'rgba(0, 0, 0, 0.6)',
  black40: 'rgba(0, 0, 0, 0.4)',
  black08: 'rgba(0, 0, 0, 0.08)',
  black15: 'rgba(0, 0, 0, 0.15)',

  // blue gradations
  blue: "#b0c4f1",
  blue40: 'rgba(176, 196, 241, 0.4)',
  blue15: 'rgba(176, 196, 241, 0.15)',
  blue8: 'rgba(176, 196, 241, 0.08)',

  // light mode
  surface: '#FFFFFF',
  background: '#F7F8FA',
  onPrimary: '#FFFFFF',
  backgroundOnPrimary: '#FFFFFF',
  backgroundSurface: '#FFFFFF',
  backgroundOnSurface: '#2D2D2D',
  primary: '#4475E3',
  primaryShadow: '#4475E361',
  primaryLight: '#ECF1FC',
  primaryHovered: '#2F65E0',
  primary05: '#F6F8FE',
  primary10: '#ECF1FC',
  primary20: '#E3EAFB',
  primary30: '#C7D6F7',
  primaryDisabled: '#B3C6F1',
  primaryLightPressed: '#E3EAFB',
  primaryLightFocused: '#C7D6F7',
  textPrimary: '#474747',
  primaryWhite: '#FFFFFF',
  disabled: '#DBDBDB',
  disabled20: 'rgba(219, 219, 219, 0.2)',
  onDisabled: '#A1A1A1',

  textPrimaryBlue: '#4475E3',
  textPrimaryBlue10: 'rgba(68, 117, 227, 0.2)',
  textPrimaryDark: '#474747',
  textSecondaryGray: '#808080',
  textDisabled: '#838383',
  onSurface: '#2D2D2D',
  secondarySkyblue: '#00B0D7', // replace
  backgoundLight: '#F0F0F0',
  primaryBlueHovered: '#2F65E0',
  primaryBluePressed: '#2057D5',

  statusSuccess: '#2DC008',
  statusSuccess10: '#DCF2EA',
  statusSuccessText: '#00825D',

  statusError: '#FF3333',
  statusError10: '#FFEBEB',
  statusErrorText: '#D14343',

  statusWarning: '#F5BF00',
  statusWarning10: '#FFF7D0',
  statusWarningText: '#A36400',

  secondaryViolet: '#944ED7',
  secondaryVioletHovered: '#6A3799',
  secondaryViolet10: '#F4EDFB',

  secondaryTangerine: '#F39C18',
  secondaryTangerineHovered: '#CC8314',
  secondaryTangerine10: '#FEF5E8',

  secondarySkyBlue: '#00B0D7',
  secondarySkyBlueHovered: '#0092B3',
  secondarySkyBlue10: '#E5F7FB',

  secondaryGreen: '#1AD598',
  secondaryGreenHovered: '#16B37F',
  secondaryGreen10: '#E8FBF5',

  secondaryRed: '#FA5F4F',
  secondaryRedHovered: '#CC4E40',
  secondaryRed10: '#FFEFED',

  transparent: 'rgba(0, 0, 0, 0)'
};

export type SpecColorType = keyof typeof colors;

const weightToCssValue = {
  regular: 400,
  upperregular: 450,
  medium: 500,
  semibold: 600,
  bold: 700,
};

const fontSpec = (
  size: number,
  lineHeight: number | string,
  weight: keyof typeof weightToCssValue,
  fontFamily: string
) => css`
  font-family: ${fontFamily}, sans-serif;
  font-size: ${px(size)};
  line-height: ${typeof lineHeight === 'string' ? lineHeight : px(lineHeight)};
  font-weight: ${weightToCssValue[weight]};
  font-style: normal;
`;

const typography = {
  // Inter typeface
  labelRegular: fontSpec(10, 15, 'regular', 'Inter'),
  labelSemibold: fontSpec(10, 15, 'semibold', 'Inter'),
  smallLabelRegular: fontSpec(8, 10, 'regular', 'Inter'),

  bodyXSmallRegular: fontSpec(12, 18, 'regular', 'Inter'),
  bodyXSmallSemibold: fontSpec(12, 18, 'semibold', 'Inter'),
  bodySmallRegular: fontSpec(14, 21, 'regular', 'Inter'),
  bodySmallSemibold: fontSpec(14, 21, 'semibold', 'Inter'),
  bodyMediumRegular: fontSpec(16, 24, 'regular', 'Inter'),
  bodyMediumSemibold: fontSpec(16, 24, 'semibold', 'Inter'),
  bodyLargeRegular: fontSpec(18, 26, 'regular', 'Inter'),
  bodyLargeSemibold: fontSpec(18, 26, 'semibold', 'Inter'),

  headingXSmall: fontSpec(16, 21, 'semibold', 'Inter'),
  headingSmall: fontSpec(18, 23, 'semibold', 'Inter'),
  headingXMedium: fontSpec(18, 27, 'semibold', 'Inter'),
  headingXMediumRegular: fontSpec(18, 27, 'regular', 'Inter'),
  headingMedium: fontSpec(24, 32.2, 'semibold', 'Inter'),
  headingLargeSemibold: fontSpec(32, 42, 'semibold', 'Inter'),
  heading2LargeSemibold: fontSpec(40, 52, 'semibold', 'Inter'),
  headingXLargeSemibold: fontSpec(48, 58, 'semibold', 'Inter'),
  headingXXLargeSemibold: fontSpec(60, 72, 'semibold', 'Inter'),

  portalDisplayLarge: fontSpec(72, 86, 'semibold', 'Inter'),

  sdkBodySmallRegular: fontSpec(14, '130%', 'regular', 'Inter'),
  sdkBodyLargeRegular: fontSpec(18, 24, 'regular', 'Inter'),
  sdkBodyLargeSemibold: fontSpec(18, 24, 'semibold', 'Inter'),
  sdkHeadingMedium: fontSpec(20, 26, 'semibold', 'Inter'),
  sdkBodyXXLRegular: fontSpec(32, 42, 'regular', 'Inter'),
  sdkBodyXXLSemibold: fontSpec(32, '130%', 'semibold', 'Inter'),
  sdkBodyMediumRegular: fontSpec(16, '130%', 'regular', 'Inter'),
  sdkBodyMediumSemibold: fontSpec(16, '130%', 'semibold', 'Inter'),
  sdkDisplayMedium: fontSpec(20, 26, 'semibold', 'Inter'),
  sdkHeadingLarge: fontSpec(40, '130%', 'semibold', 'Inter'),

  // Fira Code typeface
  query14: fontSpec(14, '130%', 'upperregular', 'Fira Code'),
  query12: fontSpec(12, '150%', 'upperregular', 'Fira Code'),
};

const boxShadow = {
  card: `0 0 ${px(2)} ${px(0)} rgba(0,0,0,0.15)`,
  avatar: `0 ${px(8)} ${px(10)} ${px(2)} 0 rgba(0,0,0,0.12)`,
  studyAvatar: `0 ${px(15)} ${px(20)} rgba(180,180,180,0.3)`,
  previewScreen: `0 ${px(2)} ${px(4)} rgba(71,71,71,0.25)`,
  skipLogicHeader: `box-shadow: ${px(3)} ${px(4)} ${px(15)} rgba(0, 0, 0, 0.05)`,
};

const animation = {
  defaultTiming: 'cubic-bezier(0.45, 0.05, 0, 1)',
  rippleTiming: 'cubic-bezier(0.4, 0, 0.2, 1)',
};

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof colors;
    typography: typeof typography;
    boxShadow: typeof boxShadow;
    animation: typeof animation;
  }
}

const theme: DefaultTheme = {
  colors,
  typography,
  boxShadow,
  animation,
};

export default theme;
