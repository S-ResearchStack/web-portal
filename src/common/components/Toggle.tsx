import React, { FC, ReactElement, useMemo } from 'react';
import styled, { keyframes } from 'styled-components';
import spring, { toString as springToString } from 'css-spring';
import _uniqueId from 'lodash/uniqueId';

import { colors, px, typography } from 'src/styles';

const UNCHECKED_HOVERED_SPRING_CONFIG = { stiffness: 256, damping: 24 };
const CHECKED_SPRING_CONFIG = { stiffness: 711.1, damping: 40 };

const uncheckedHoverAnimation = keyframes`
  ${springToString(spring({ left: px(2) }, { left: px(4) }, UNCHECKED_HOVERED_SPRING_CONFIG))}
`;

const checkedAnimation = keyframes`
  ${springToString(spring({ left: px(4) }, { left: px(18) }, CHECKED_SPRING_CONFIG))}
`;

interface ToggleProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: ReactElement | string;
}

const ANIMATION_FN = 'cubic-bezier(0, 0, 0.58, 1)';

const ToggleContainer = styled.div`
  padding-left: ${px(2)};

  input {
    display: none;

    & + label {
      height: ${px(40)};
      min-width: ${px(40)};
      position: relative;
      display: flex;
      align-items: center;
      gap: ${px(12)};

      &:after,
      &:before {
        content: '';
        display: block;
        position: absolute;
        animation-fill-mode: forwards;
      }

      &:before {
        width: ${px(36)};
        height: ${px(20)};
        border-radius: ${px(10)};
        background-color: ${colors.disabled};
        transition: background-color 300ms ${ANIMATION_FN};
      }

      &:after {
        width: ${px(16)};
        height: ${px(16)};
        border-radius: ${px(8)};
        background-color: ${colors.backgroundOnPrimary};
        left: ${px(2)};
      }

      > span {
        ${typography.bodySmallRegular};
        color: ${colors.textPrimary};
        margin-left: ${px(50)};
        transition: color 300ms ${ANIMATION_FN};
      }
    }

    &:enabled + label:hover {
      cursor: pointer;
    }

    &:not(:checked):not(:disabled) + label:hover {
      &:after {
        animation-name: ${uncheckedHoverAnimation};
        animation-duration: 500ms;
      }
    }

    &:checked + label {
      &:before {
        background-color: ${colors.primary};
      }

      &:after {
        animation-name: ${checkedAnimation};
        animation-duration: 500ms;
      }

      &:hover {
        &:before {
          background-color: ${colors.primaryHovered};
        }
      }
    }

    &:disabled {
      & + label {
        &:before {
          background-color: ${colors.disabled};
        }

        &:after {
          background-color: ${colors.onDisabled};
        }

        > span {
          color: ${colors.textDisabled};
        }
      }

      &:checked + label {
        &:before {
          background-color: ${colors.primaryDisabled};
        }

        &:after {
          background-color: ${colors.backgroundOnPrimary};
        }
      }
    }
  }
`;

const Toggle: FC<ToggleProps> = ({ className, label, ...props }) => {
  const id = useMemo(() => _uniqueId('toggle_'), []);

  return (
    <ToggleContainer className={className}>
      <input {...props} id={id} type="checkbox" />
      <label data-testid="label" htmlFor={id}>
        {label && <span>{label}</span>}
      </label>
    </ToggleContainer>
  );
};

export default Toggle;
