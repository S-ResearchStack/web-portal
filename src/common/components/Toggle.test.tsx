import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/react';
import theme from 'src/styles/theme';
import { ThemeProvider } from 'styled-components/';
import Toggle from './Toggle';

describe('Toggle', () => {
  it('test toggle click', async () => {
    const { baseElement, getByTestId, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Toggle data-testid="toggle" label="toggle" />
      </ThemeProvider>
    );

    const toggle = getByTestId('toggle');
    const label = queryByTestId('label') as Element;

    expect(baseElement).toMatchSnapshot();
    expect(toggle).not.toBeChecked();

    await userEvent.click(toggle);

    expect(baseElement).toMatchSnapshot();
    expect(toggle).toBeChecked();

    await userEvent.click(label);

    expect(toggle).not.toBeChecked();
  });

  it('[NEGATIVE] test toggle disabled state', async () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Toggle data-testid="toggle" label="toggle" disabled />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
    expect(getByTestId('toggle')).toHaveAttribute('disabled');
  });

  it('[NEGATIVE] should prevent change checked state while element is disabled', async () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Toggle data-testid="toggle" label="toggle" disabled />
      </ThemeProvider>
    );

    const toggle = getByTestId('toggle');

    expect(baseElement).toMatchSnapshot();
    expect(getByTestId('toggle')).toHaveAttribute('disabled');
    expect(toggle).not.toBeChecked();

    await userEvent.click(toggle);

    expect(toggle).not.toBeChecked();
  });

  it('[NEGATIVE] should render with wrong props', async () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Toggle
          data-testid="toggle"
          label={['toggle-test'] as unknown as string}
          checked={'unknown' as unknown as boolean}
          onChange={() => {}}
        />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
    expect(getByTestId('toggle')).toBeChecked();
    expect(getByTestId('label')).toHaveTextContent('toggle-test');
  });
});
