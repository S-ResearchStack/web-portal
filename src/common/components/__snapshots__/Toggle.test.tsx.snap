// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle [NEGATIVE] should prevent change checked state while element is disabled 1`] = `
.c0 {
  padding-left: 0.125rem;
}

.c0 input {
  display: none;
}

.c0 input + label {
  height: 2.5rem;
  min-width: 2.5rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75rem;
}

.c0 input + label:after,
.c0 input + label:before {
  content: '';
  display: block;
  position: absolute;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c0 input + label:before {
  width: 2.25rem;
  height: 1.25rem;
  border-radius: 0.625rem;
  background-color: #DBDBDB;
  -webkit-transition: background-color 300ms cubic-bezier(0,0,0.58,1);
  transition: background-color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input + label:after {
  width: 1rem;
  height: 1rem;
  border-radius: 0.5rem;
  background-color: #FFFFFF;
  left: 0.125rem;
}

.c0 input + label > span {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-left: 3.125rem;
  -webkit-transition: color 300ms cubic-bezier(0,0,0.58,1);
  transition: color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input:enabled + label:hover {
  cursor: pointer;
}

.c0 input:not(:checked):not(:disabled) + label:hover:after {
  -webkit-animation-name: XQEqq;
  animation-name: XQEqq;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:before {
  background-color: #4475E3;
}

.c0 input:checked + label:after {
  -webkit-animation-name: gzQhmP;
  animation-name: gzQhmP;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:hover:before {
  background-color: #2F65E0;
}

.c0 input:disabled + label:before {
  background-color: #DBDBDB;
}

.c0 input:disabled + label:after {
  background-color: #A1A1A1;
}

.c0 input:disabled + label > span {
  color: #838383;
}

.c0 input:disabled:checked + label:before {
  background-color: #B3C6F1;
}

.c0 input:disabled:checked + label:after {
  background-color: #FFFFFF;
}

<body>
  <div>
    <div
      class="c0"
    >
      <input
        data-testid="toggle"
        disabled=""
        id="toggle_1"
        type="checkbox"
      />
      <label
        data-testid="label"
        for="toggle_1"
      >
        <span>
          toggle
        </span>
      </label>
    </div>
  </div>
</body>
`;

exports[`Toggle [NEGATIVE] should render with wrong props 1`] = `
.c0 {
  padding-left: 0.125rem;
}

.c0 input {
  display: none;
}

.c0 input + label {
  height: 2.5rem;
  min-width: 2.5rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75rem;
}

.c0 input + label:after,
.c0 input + label:before {
  content: '';
  display: block;
  position: absolute;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c0 input + label:before {
  width: 2.25rem;
  height: 1.25rem;
  border-radius: 0.625rem;
  background-color: #DBDBDB;
  -webkit-transition: background-color 300ms cubic-bezier(0,0,0.58,1);
  transition: background-color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input + label:after {
  width: 1rem;
  height: 1rem;
  border-radius: 0.5rem;
  background-color: #FFFFFF;
  left: 0.125rem;
}

.c0 input + label > span {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-left: 3.125rem;
  -webkit-transition: color 300ms cubic-bezier(0,0,0.58,1);
  transition: color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input:enabled + label:hover {
  cursor: pointer;
}

.c0 input:not(:checked):not(:disabled) + label:hover:after {
  -webkit-animation-name: XQEqq;
  animation-name: XQEqq;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:before {
  background-color: #4475E3;
}

.c0 input:checked + label:after {
  -webkit-animation-name: gzQhmP;
  animation-name: gzQhmP;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:hover:before {
  background-color: #2F65E0;
}

.c0 input:disabled + label:before {
  background-color: #DBDBDB;
}

.c0 input:disabled + label:after {
  background-color: #A1A1A1;
}

.c0 input:disabled + label > span {
  color: #838383;
}

.c0 input:disabled:checked + label:before {
  background-color: #B3C6F1;
}

.c0 input:disabled:checked + label:after {
  background-color: #FFFFFF;
}

<body>
  <div>
    <div
      class="c0"
    >
      <input
        checked=""
        data-testid="toggle"
        id="toggle_1"
        type="checkbox"
      />
      <label
        data-testid="label"
        for="toggle_1"
      >
        <span>
          toggle-test
        </span>
      </label>
    </div>
  </div>
</body>
`;

exports[`Toggle [NEGATIVE] test toggle disabled state 1`] = `
.c0 {
  padding-left: 0.125rem;
}

.c0 input {
  display: none;
}

.c0 input + label {
  height: 2.5rem;
  min-width: 2.5rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75rem;
}

.c0 input + label:after,
.c0 input + label:before {
  content: '';
  display: block;
  position: absolute;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c0 input + label:before {
  width: 2.25rem;
  height: 1.25rem;
  border-radius: 0.625rem;
  background-color: #DBDBDB;
  -webkit-transition: background-color 300ms cubic-bezier(0,0,0.58,1);
  transition: background-color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input + label:after {
  width: 1rem;
  height: 1rem;
  border-radius: 0.5rem;
  background-color: #FFFFFF;
  left: 0.125rem;
}

.c0 input + label > span {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-left: 3.125rem;
  -webkit-transition: color 300ms cubic-bezier(0,0,0.58,1);
  transition: color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input:enabled + label:hover {
  cursor: pointer;
}

.c0 input:not(:checked):not(:disabled) + label:hover:after {
  -webkit-animation-name: XQEqq;
  animation-name: XQEqq;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:before {
  background-color: #4475E3;
}

.c0 input:checked + label:after {
  -webkit-animation-name: gzQhmP;
  animation-name: gzQhmP;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:hover:before {
  background-color: #2F65E0;
}

.c0 input:disabled + label:before {
  background-color: #DBDBDB;
}

.c0 input:disabled + label:after {
  background-color: #A1A1A1;
}

.c0 input:disabled + label > span {
  color: #838383;
}

.c0 input:disabled:checked + label:before {
  background-color: #B3C6F1;
}

.c0 input:disabled:checked + label:after {
  background-color: #FFFFFF;
}

<body>
  <div>
    <div
      class="c0"
    >
      <input
        data-testid="toggle"
        disabled=""
        id="toggle_1"
        type="checkbox"
      />
      <label
        data-testid="label"
        for="toggle_1"
      >
        <span>
          toggle
        </span>
      </label>
    </div>
  </div>
</body>
`;

exports[`Toggle test toggle click 1`] = `
.c0 {
  padding-left: 0.125rem;
}

.c0 input {
  display: none;
}

.c0 input + label {
  height: 2.5rem;
  min-width: 2.5rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75rem;
}

.c0 input + label:after,
.c0 input + label:before {
  content: '';
  display: block;
  position: absolute;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c0 input + label:before {
  width: 2.25rem;
  height: 1.25rem;
  border-radius: 0.625rem;
  background-color: #DBDBDB;
  -webkit-transition: background-color 300ms cubic-bezier(0,0,0.58,1);
  transition: background-color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input + label:after {
  width: 1rem;
  height: 1rem;
  border-radius: 0.5rem;
  background-color: #FFFFFF;
  left: 0.125rem;
}

.c0 input + label > span {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-left: 3.125rem;
  -webkit-transition: color 300ms cubic-bezier(0,0,0.58,1);
  transition: color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input:enabled + label:hover {
  cursor: pointer;
}

.c0 input:not(:checked):not(:disabled) + label:hover:after {
  -webkit-animation-name: XQEqq;
  animation-name: XQEqq;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:before {
  background-color: #4475E3;
}

.c0 input:checked + label:after {
  -webkit-animation-name: gzQhmP;
  animation-name: gzQhmP;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:hover:before {
  background-color: #2F65E0;
}

.c0 input:disabled + label:before {
  background-color: #DBDBDB;
}

.c0 input:disabled + label:after {
  background-color: #A1A1A1;
}

.c0 input:disabled + label > span {
  color: #838383;
}

.c0 input:disabled:checked + label:before {
  background-color: #B3C6F1;
}

.c0 input:disabled:checked + label:after {
  background-color: #FFFFFF;
}

<body>
  <div>
    <div
      class="c0"
    >
      <input
        data-testid="toggle"
        id="toggle_1"
        type="checkbox"
      />
      <label
        data-testid="label"
        for="toggle_1"
      >
        <span>
          toggle
        </span>
      </label>
    </div>
  </div>
</body>
`;

exports[`Toggle test toggle click 2`] = `
.c0 {
  padding-left: 0.125rem;
}

.c0 input {
  display: none;
}

.c0 input + label {
  height: 2.5rem;
  min-width: 2.5rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75rem;
}

.c0 input + label:after,
.c0 input + label:before {
  content: '';
  display: block;
  position: absolute;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c0 input + label:before {
  width: 2.25rem;
  height: 1.25rem;
  border-radius: 0.625rem;
  background-color: #DBDBDB;
  -webkit-transition: background-color 300ms cubic-bezier(0,0,0.58,1);
  transition: background-color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input + label:after {
  width: 1rem;
  height: 1rem;
  border-radius: 0.5rem;
  background-color: #FFFFFF;
  left: 0.125rem;
}

.c0 input + label > span {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-left: 3.125rem;
  -webkit-transition: color 300ms cubic-bezier(0,0,0.58,1);
  transition: color 300ms cubic-bezier(0,0,0.58,1);
}

.c0 input:enabled + label:hover {
  cursor: pointer;
}

.c0 input:not(:checked):not(:disabled) + label:hover:after {
  -webkit-animation-name: XQEqq;
  animation-name: XQEqq;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:before {
  background-color: #4475E3;
}

.c0 input:checked + label:after {
  -webkit-animation-name: gzQhmP;
  animation-name: gzQhmP;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
}

.c0 input:checked + label:hover:before {
  background-color: #2F65E0;
}

.c0 input:disabled + label:before {
  background-color: #DBDBDB;
}

.c0 input:disabled + label:after {
  background-color: #A1A1A1;
}

.c0 input:disabled + label > span {
  color: #838383;
}

.c0 input:disabled:checked + label:before {
  background-color: #B3C6F1;
}

.c0 input:disabled:checked + label:after {
  background-color: #FFFFFF;
}

<body>
  <div>
    <div
      class="c0"
    >
      <input
        data-testid="toggle"
        id="toggle_1"
        type="checkbox"
      />
      <label
        data-testid="label"
        for="toggle_1"
      >
        <span>
          toggle
        </span>
      </label>
    </div>
  </div>
</body>
`;
